### ISSUE TITLE
`pmpm` uses the `node` version it was installed it instead of currently active/pinned version

### ISSUE THREAD

**Issue by nifr** [notion-cli/notion#1562]
**Given** volta's active node version is i.e. `v20.8.1`
**When** I install `pnpm` with `volta`
**And** I install and pin a different node version i.e. `v21.0.0`
**Then** `pnpm` still uses the previous `node` version `v20.8.1`

Steps to reproduce:

```
volta install node@20.8.1
volta install pnpm@8.9.2
volta install node@21.0.0
volta pin node@21.0.0
pnpm run dev
```

With `package.json` containing:

```
  "packageManager": "pnpm@8.9.2",
  "volta": {
    "node": "21.0.0"
  },
  "engines": {
    "node": "21.0.0",
    "pnpm":"8.9.2"
  }
 ```
 
 The following warning appears and reveals that the incorrect `node` version is used under the hood:

```
WARN  Unsupported engine: wanted: {"node":"v21.0.0"} (current: {"node":"v20.8.1","pnpm":"8.9.2"})
```

**Comment by lananelson** [comment]
+1 and I cannot uninstall Node that volta installed

**Comment by chawyehsu** [comment]
I can't reproduce with steps that OP posted. I may be able to take this issue on but I need a certainly reproducible way to reproduce.

```
volta install node@20.8.1
volta install pnpm@8.9.2
volta install node@21.0.0
volta pin node@21.0.0
volta pin pnpm@8.9.2
pnpm run dev
```

```json
{
  "name": "test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "dev": "node index.js"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "volta": {
    "node": "21.0.0",
    "pnpm": "8.9.2"
  }
}
```

**Comment by theodesp** [comment]
> VOLTA_FEATURE_PNPM=1
I got this working with this export:

❯ pnpm install
 WARN  Unsupported engine: wanted: {"node":">=18"} (current: {"node":"v16.20.0","pnpm":"8.6.2"})
Lockfile is up to date, resolution step is skipped
Already up to date

> datree@0.0.1 prepare /Users/theo.despoudis/Workspace/datree
> husky install

husky - Git hooks installed
Done in 1.2s
❯ export VOLTA_FEATURE_PNPM=1
❯ pnpm install
Lockfile is up to date, resolution step is skipped
Already up to date

> datree@0.0.1 prepare /Users/theo.despoudis/Workspace/datree
> husky install

husky - Git hooks installed
Done in 1.1s

**Comment by chawyehsu** [comment]
> I also experience the same issue with the `pnpm@7.33.6` and `volta@1.0.5`.
> 
> The project has declared the `node@18.12.1`, shown in the picture below, with `node -v`. While executing the `pnpm node -v`, the node version used was `node@18.19.0`, which was the node version used when installing the `pnpm`.
> 
> Let's say we run the NextJS project with the `pnpm dev`; then the node version is attached to the `pnpm`, not the one declared in the project itself.
> 
> I have also tried adding the global environment variable stated in https://docs.volta.sh/advanced/pnpm; it looks like the feature is still experimental.
> 
> I would appreciate it if someone could help with this issue; thank you!
> 
> ![image](https://private-user-images.githubusercontent.com/25271386/297302630-83cbf7bd-a705-4149-bcf6-c545a01e7a1a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU0ODUyMjUsIm5iZiI6MTcwNTQ4NDkyNSwicGF0aCI6Ii8yNTI3MTM4Ni8yOTczMDI2MzAtODNjYmY3YmQtYTcwNS00MTQ5LWJjZjYtYzU0NWEwMWU3YTFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE3VDA5NDg0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE0NGRkMTgxMmVkMTFlYjJjZjI2MDg5N2JhMWM0MTQwZjExZjMwZjY4MDVjZTFkOTExMGE5Yzc2ZjljZWY1NDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.oH2MFT3Vmw3uPMPkml4ShE0Q8AohsFos0mi4562_7Ug)

You are using volta v1.0.5 while the `VOLTA_FEATURE_PNPM` feature support was introduced in Volta [v1.1.1](https://github.com/volta-cli/volta/releases/tag/v1.1.1)...

**Comment by malfianrasyidin** [comment]
> > I also experience the same issue with the `pnpm@7.33.6` and `volta@1.0.5`.
> > The project has declared the `node@18.12.1`, shown in the picture below, with `node -v`. While executing the `pnpm node -v`, the node version used was `node@18.19.0`, which was the node version used when installing the `pnpm`.
> > Let's say we run the NextJS project with the `pnpm dev`; then the node version is attached to the `pnpm`, not the one declared in the project itself.
> > I have also tried adding the global environment variable stated in https://docs.volta.sh/advanced/pnpm; it looks like the feature is still experimental.
> > I would appreciate it if someone could help with this issue; thank you!
> > ![image](https://private-user-images.githubusercontent.com/25271386/297302630-83cbf7bd-a705-4149-bcf6-c545a01e7a1a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU0ODUyMjUsIm5iZiI6MTcwNTQ4NDkyNSwicGF0aCI6Ii8yNTI3MTM4Ni8yOTczMDI2MzAtODNjYmY3YmQtYTcwNS00MTQ5LWJjZjYtYzU0NWEwMWU3YTFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE3VDA5NDg0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE0NGRkMTgxMmVkMTFlYjJjZjI2MDg5N2JhMWM0MTQwZjExZjMwZjY4MDVjZTFkOTExMGE5Yzc2ZjljZWY1NDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.oH2MFT3Vmw3uPMPkml4ShE0Q8AohsFos0mi4562_7Ug)
> 
> You are using volta v1.0.5 while the `VOLTA_FEATURE_PNPM` feature support was introduced in Volta [v1.1.1](https://github.com/volta-cli/volta/releases/tag/v1.1.1)...

You are right, @chawyehsu, thank you. The docs should have mentioned the minimum version of the Volta itself.

**Comment by olja-ivkovic** [comment]
I had the same issue even with Volta v1.1.1. and VOLTA_FEATURE_PNPM flag enabled, but managed to fix it by uninstalling `pnpm` and installing it again while node 20.11.0 was pinned.

So just:
```
volta uninstall pnpm
```

```
volta pin node@20.11.0
```
( or volta install node@20.11.0 to have it as a default node version)

```
volta install pnpm
```

And now this command finally tells me that pnpm uses my desired node version:
```
pnpm node --version
```

```
> 20.11.0
```

**Comment by mfwgenerics** [comment]
Seeing this too. The uninstall/reinstall workaround works for me.

**Comment by 6525146** [comment]
Alla is grate

**Comment by mb21** [comment]
Uh, that was indeed a gotcha. I've made a PR to at least update the docs regarding this known bug: https://github.com/volta-cli/website/pull/139/files
